app/models/like.rb

```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :post, counter_cache: true
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_likes_count
```

Comment:
The same as above:

[OPTIONAL] It would be nice if you use `after_save` method for updating likes count and make `update_likes_count` private. More info: [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)